packages/firestore/src/core/sync_engine.ts

```diff
@@ -246,8 +252,7 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
       }
     }
 
-    this.syncEngineListener!.onWatchChange([viewSnapshot]);
```

Comment:
I had to remove this indirection. `onWatchChange` doesn't fire if EventManager doesn't execute `this.queries.set(query, queryInfo)`. Executing `this.queries.set(query, queryInfo!)` before this method runs breaks listeners if the initial registration in IndexedDB failed.

Reply:
At first I was concerned about this because it seemed like this would prevent calling `onViewSnapshot` on the listener, but actually it looks like code to do that was already explicitly in the `EventManager` (in `if (queryInfo.viewSnap)`, and I'm guessing that the explicit code was just being treated as a no-op because previously `queryInfo.viewSnap` would be undefined before?

Reply:
I am not sure I fully understand this, but I hope that my previous comment reliefs your concerns. Let me know if it doesn't.